packages/eslint-plugin-template/src/rules/i18n.ts

```diff
@@ -177,7 +177,7 @@ export default createESLintRule<Options, MessageIds>({
     const parserServices = getTemplateParserServices(context);
     const sourceCode = context.getSourceCode();
     const allowedBoundTextPattern = RegExp(
-      boundTextAllowedPattern ?? DEFAULT_BOUND_TEXT_ALLOWED_PATTERN,
+      boundTextAllowedPattern ?? DEFAULT_ALLOWED_BOUND_TEXT_PATTERN,
```

Comment:
Please can we avoid passing an existing instance of RegExp to the RegExp constructor here? It is only needed for the string coming from configuration.

Additionally please can you add the `new` keyword to the RegExp constructor call, it works without it but I think it is preferable to have it there.

Reply:
> Please can we avoid passing an existing instance of RegExp to the RegExp constructor here? It is only needed for the string coming from configuration.

Oh yeah, for sure, I missed that!

> Additionally please can you add the new keyword to the RegExp constructor call, it works without it because of legacy reasons but I think it is preferable to have it there.

_"because of legacy reasons"_ — Is there any reference where I can find this? Also, is it something related to `RegExp(...)` specifically or, for example, `Error(...)` too? To clarify, I'm not doubting or anything like that, just asking because I've never seen that we shouldn't use it like this because it's legacy and somewhat will be deprecated. I thought it just to work like the the other factories `Boolean(...)`, `String(...)`or even `Error(...)` instead of calling the classes directly `new Boolean(...)`, `new String(...)`, `new Error(...)`.

That being said, of course I can change this without any problem.

Reply:
Sorry for not being clearer. What I mean by that is:

Constructors functions such as `RegExp` and `Error` come from a time before the ECMAScript class feature was introduced. They are _functionally_ classes and were they introduced as globals today, they would literally be classes, however classes have different semantics to straight functions so they can never be converted - that would break the web.

As I'm sure you know, when you use a class, you cannot use its constructor without the `new` keyword, you will get a `TypeError`. Were it not for the legacy of the web, these globals would therefore also throw a `TypeError` without the use of the `new` keyword, that's what I meant by my comment above.

Completely anecdotal, but I _feel_ like I see far more cases of `RegExp` and `Error` used with the `new` keyword than without (the same is true for `Array`). Interestingly I feel the opposite about those representing non-object primitive values - `String`, `Boolean`, `Number`. I feel like I never/almost never see those use with `new`.

A fine old mess :D

Reply:
Thanks, it's clear now.

is really interesting, taking a look at the `@angular/components` codebase, for example, I find a lot more `Error(...)` instead of `new Error(...)`, while a lot more `new RegExp(...)` than `RegExp(...)` 😅